components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/OIDCClaimUtil.java

```diff
@@ -270,4 +279,98 @@ private static boolean isUserConsentRequiredForClaims(String grantType) {
 
         return OAuthServerConfiguration.getInstance().isUserConsentRequiredForClaims(grantType);
     }
+
+    /**
+     * Get the application's preferred subject type.
+     *
+     * @param clientId client id
+     * @return subject type
+     */
+    private static SubjectType getSubjectType(String clientId) throws IdentityOAuth2Exception {
+
+        ServiceProvider serviceProvider = OAuth2Util.getServiceProvider(clientId);
+        ServiceProviderProperty[] serviceProviderProperties = serviceProvider.getSpProperties();
+        try {
+            for (ServiceProviderProperty serviceProviderProperty : serviceProviderProperties) {
+                if (SUBJECT_TYPE.equals(serviceProviderProperty.getName())) {
+                    return SubjectType.fromValue(serviceProviderProperty.getValue());
+                }
+            }
+        } catch (IllegalArgumentException e) {
+            // return default subject type if an incorrect value is configured.
+            return StringUtils.isNotBlank(OAuthServerConfiguration.getInstance().getDefaultSubjectType()) ? SubjectType
```

Comment:
Throw the exception

Reply:
fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/commit/c0be30590afde90f5f4e8c291f631740e9cc1326